probe_scraper/transform_probes.py

```diff
@@ -189,68 +190,85 @@ def transform(probe_data, node_data, break_by_channel):
     return result_data
 
 
-def make_commit_hash_probe_definition(definition, commit):
+def make_commit_hash_metric_definition(definition, commit, commit_timestamps):
     if COMMITS_KEY not in definition:
         # This is the first time we've seen this definition
         definition[COMMITS_KEY] = {
             "first": commit,
             "last": commit
         }
+        definition[DATES_KEY] = {
+            "first": commit_timestamps[commit],
+            "last": commit_timestamps[commit]
+        }
     else:
         # we've seen this definition, update the `last` commit
         definition[COMMITS_KEY]["last"] = commit
+        definition[DATES_KEY]["last"] = commit_timestamps[commit]
 
     return definition
 
 
-def update_or_add_probe(all_probes, repo_name, commit_hash, probe, probe_type, definition):
-    probe_id = get_probe_id(probe_type, probe)
+def metrics_equal(def1, def2):
+    return all((
```

Comment:
I think because these are `dataclasses`, you can just do `def1 == def2`.  Did you try that and it didn't work?

Reply:
No, I didn't! That sounds fantastic.

Reply:
This didn't work because I'm calling `.serialize()` on the objects before it gets to this point :/. What would be ideal, I think, is to extend those classes with ones that include the new fields, add a comparator in that class, and use that here.

In general though this library could use a lot of refactoring...

Reply:
This solution here is fine given all that...

Reply:
I've opened https://github.com/mozilla/probe-scraper/issues/74 to call attention to a refactoring effort. If you find some time it may be helpful to write some thoughts on what you think a good refactor would look like here.